calculator.rb

```diff
@@ -0,0 +1,83 @@
+puts "Welcome to the Ruby Calculator!"
+
+operator = ""
+# valid operators
+operators = [
+  "add", "+", "1", 
+  "subtract", "-", "2", 
+  "multiply", "*", "3", 
+  "divide", "/", "4",
+  "exponent", "**", "5",
+  "modulo", "%", "6"
+]
+
+# prompt user for operator until they enter a valid option
+until operators.include?(operator)
+  puts "Please choose an operation from the list (name, symbol, or number):
+  1. add(+)
+  2. subtract(-)
+  3. multiply(*)
+  4. divide(/)
+  5. exponent(**)
+  6. modulo(%)"
+  operator = gets.chomp.to_s.downcase
+end
+
+# update operator to operation to use in user prompt
+case operator
+when "add", "+", "1"
+  operator = "addition"
+when "subtract", "-", "2"
+  operator = "subtraction"
+when "multiply", "*", "3"
+  operator = "multiplication"
+when "divide", "/", "4"
+  operator = "division"
+when "exponent", "**", "5"
+  operator = "exponent"
+when "modulo", "%", "6"
+  operator = "modulo"
+end
+
+# validate that user input is numeric
+def validate_number(num)
```

Comment:
I suggest moving your methods to the top of the file for now, later another file, and putting the main driver code below.